datacube/utils/geometry.py

```diff
@@ -219,8 +219,12 @@ def __repr__(self):
     def __eq__(self, other):
         if isinstance(other, compat.string_types):
             other = CRS(other)
-        canonical = lambda crs: set(crs.ExportToProj4().split() + ['+wktext'])
-        return canonical(self._crs) == canonical(other._crs)  # pylint: disable=protected-access
+        gdal_thinks_issame = self._crs.IsSame(other._crs) == 1  # pylint: disable=protected-access
```

Comment:
why not early exit?

``` python
if self._crs.IsSame(other._crs) == 1:  # trust gdal assesment
    return True
```

Reply:
No particular reason, just trying to be self documenting.

Reply:
It all depends on how costly is computation and how often it is done, no point computing the second one if the first one is `True`. Building a string, splitting it into an array and constructing a set out of that, times 2 is a bit of work, but I guess it isn't being done a lot.